src/main/java/uk/gov/hmcts/ccd/data/caseaccess/CaseAssignedUserRolesKey.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.ccd.data.caseaccess;
+
+import java.util.List;
+import org.apache.commons.collections.ListUtils;
+
+public class CaseAssignedUserRolesKey {
+
+    private final List<Long> caseIds;
+
+    private final List<String> userIds;
+
+    public CaseAssignedUserRolesKey(List<Long> caseIds, List<String> userIds) {
+        this.caseIds = caseIds;
+        this.userIds = userIds;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        CaseAssignedUserRolesKey that = (CaseAssignedUserRolesKey) o;
+        if (caseIds == null || that.caseIds == null || caseIds.size() != that.caseIds.size()) {
+            return false;
+        }
+
+        if (userIds == null || that.userIds == null || userIds.size() != that.userIds.size()) {
+            return false;
+        }
+        return caseIds.containsAll(that.caseIds) && userIds.containsAll(that.userIds);
```

Comment:
But even if we do, a set equivalence should be used for this kind of a key design. What if the a list contains a duplicate entry while all others being the same as the other? I suspect we need a caching and therefore a key design like this.